edge-hub/src/Microsoft.Azure.Devices.Routing.Core/Route.cs

```diff
@@ -71,12 +77,12 @@ public override int GetHashCode()
                 hash = hash * 31 + this.Id.GetHashCode();
                 hash = hash * 31 + this.Condition.GetHashCode();
                 hash = hash * 31 + this.Source.GetHashCode();
-                hash = this.Endpoints.Aggregate(hash, (acc, b) => acc * 31 + b.GetHashCode());
+                hash = hash * 31 + this.Endpoint.GetHashCode();
                 return hash;
             }
         }
 
         public override string ToString() =>
-            string.Format(CultureInfo.InvariantCulture, "Route(\"{0}\", {1}, \"{2}\" => ({3})", this.Id, this.Source, this.Condition, string.Join(", ", this.Endpoints));
+            string.Format(CultureInfo.InvariantCulture, "Route(\"{0}\", {1}, \"{2}\" => ({3})", this.Id, this.Source, this.Condition, this.Endpoint);
```

Comment:
btw only ConsoleEndpoint and NullEndpoint implemented ToString().  Please create a bug for tracking if we want to log Endpoint details.

Reply:
I think it's fine for now, it's been this way forever and I don't think we had any trouble diagnosing logs around endpoints.  If we come back and update the Endpoint class, then we can prob opportunistically improve it then.

---
In reply to: [377446707](https://github.com/Azure/iotedge/pull/2469#discussion_r377446707) [](ancestors = 377446707)